ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+int MMAddDBFRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         struct MiraMonFeature *hMMFeature);
+int MMAddPointRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                           struct MiraMonFeature *hMMFeature,
+                           MM_INTERNAL_FID nElemCount);
+int MMAddArcRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         struct MiraMonFeature *hMMFeature,
+                         MM_INTERNAL_FID nElemCount, struct MM_AH *pArcHeader);
+int MMAddNodeRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                          MM_INTERNAL_FID nElemCount,
+                          struct MM_NH *pNodeHeader);
+int MMAddPolygonRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                             struct MiraMonFeature *hMMFeature,
+                             MM_INTERNAL_FID nElemCount,
+                             MM_N_VERTICES_TYPE nVerticesCount,
+                             struct MM_PH *pPolHeader);
+int MMCloseMMBD_XP(struct MiraMonVectLayerInfo *hMiraMonLayer);
+void MMDestroyMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer);
+
+/* -------------------------------------------------------------------- */
+/*      Managing errors and warnings                                    */
+/* -------------------------------------------------------------------- */
+
+void MMCPLError(int level, int code, const char *message)
+{
+#ifdef GDAL_COMPILATION
+    CPLError(level, code, "%s", message);
+#else
+    sprintf(local_message, "level(%d), code(%d)", level, code);
+    InfoMsg(local_message);
+    ErrorMsg(message);
+#endif
+}
+
+void MMCPLWarning(int level, int code, const char *message)
+{
+#ifdef GDAL_COMPILATION
+    CPLError(level, code, "%s", message);
+#else
+    sprintf(local_message, "level(%d), code(%d)", level, code);
+    InfoMsg(local_message);
+    InfoMsg(message);
+#endif
+}
+
+void MMCPLDebug(const char *c, const char *message)
+{
+#ifdef GDAL_COMPILATION
+    CPLDebug(c, "%s", message);
+#else
+    sprintf(local_message, "Code(%s)\n", c);
+    printf(local_message); /*ok*/
+    printf(message);       /*ok*/
+    printf("\n");          /*ok*/
+#endif
+}
+
+#ifndef GDAL_COMPILATION
+int snprintf(char *str, size_t size, const char *format, ...)
+{
+    int result;
+    va_list args;
+
+    va_start(args, format);
+    result = vsnprintf(str, size, format, args);
+    va_end(args);
+
+    return result;
+}
+#endif
+
+// Checks for potential arithmetic overflow when performing multiplication
+// operations between two GUInt64 values and converting the result to size_t.
+// Important for 32 vs. 64 bit compiling compatibility.
+int MMCheckSize_t(GUInt64 nCount, GUInt64 nSize)
+{
+    size_t nMul;
+    if ((size_t)nCount != nCount)
+        return 1;
+
+    if ((size_t)nSize != nSize)
+        return 1;
+
+    nMul = (size_t)(nCount * nSize);
+    if (nCount != 0 && nMul / nCount != nSize)
```

Comment:
The GDAL saniter CI config and OSS-Fuzz run with unsigned integer overflow considered as an error, to maximize misusage detection, so it is better to avoid relying on it with:
```suggestion
    if (nCount != 0 && nMul > SIZE_T_MAX / nCount)
```

Reply:
This has been resolved